[2025-02-17T19:46:39.678+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-17T19:46:39.690+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: kafka_streaming_process_pipeline.example_consumer manual__2025-02-17T19:46:22.248865+00:00 [queued]>
[2025-02-17T19:46:39.697+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: kafka_streaming_process_pipeline.example_consumer manual__2025-02-17T19:46:22.248865+00:00 [queued]>
[2025-02-17T19:46:39.697+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2025-02-17T19:46:39.711+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): example_consumer> on 2025-02-17 19:46:22.248865+00:00
[2025-02-17T19:46:39.717+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'kafka_streaming_process_pipeline', 'example_consumer', 'manual__2025-02-17T19:46:22.248865+00:00', '--job-id', '71', '--raw', '--subdir', 'DAGS_FOLDER/kafka_streaming_process.py', '--cfg-path', '/tmp/tmpld4yr54q']
[2025-02-17T19:46:39.719+0000] {standard_task_runner.py:105} INFO - Job 71: Subtask example_consumer
[2025-02-17T19:46:39.719+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=942) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-02-17T19:46:39.720+0000] {standard_task_runner.py:72} INFO - Started process 943 to run task
[2025-02-17T19:46:39.751+0000] {task_command.py:467} INFO - Running <TaskInstance: kafka_streaming_process_pipeline.example_consumer manual__2025-02-17T19:46:22.248865+00:00 [running]> on host c20563c2c1e7
[2025-02-17T19:46:39.812+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='kafka_streaming_process_pipeline' AIRFLOW_CTX_TASK_ID='example_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-02-17T19:46:22.248865+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-02-17T19:46:22.248865+00:00'
[2025-02-17T19:46:39.813+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-17T19:46:39.832+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2025-02-17T19:46:39.833+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077:7077 --conf spark.executor.memory=2g --executor-cores 1 --executor-memory 2g --driver-memory 1g --name arrow-spark --deploy-mode client /usr/local/app/spark_applications/user_activity_consuming.py
[2025-02-17T19:46:39.871+0000] {spark_submit.py:641} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-02-17T19:46:42.210+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SparkContext: Running Spark version 3.5.4
[2025-02-17T19:46:42.211+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SparkContext: OS info Linux, 5.15.167.4-microsoft-standard-WSL2, amd64
[2025-02-17T19:46:42.211+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SparkContext: Java version 17.0.14
[2025-02-17T19:46:42.254+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-02-17T19:46:42.318+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO ResourceUtils: ==============================================================
[2025-02-17T19:46:42.319+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-02-17T19:46:42.319+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO ResourceUtils: ==============================================================
[2025-02-17T19:46:42.320+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SparkContext: Submitted application: consumeUserActivity
[2025-02-17T19:46:42.335+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 2048, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-02-17T19:46:42.344+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO ResourceProfile: Limiting resource is cpus at 1 tasks per executor
[2025-02-17T19:46:42.346+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-02-17T19:46:42.383+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SecurityManager: Changing view acls to: default
[2025-02-17T19:46:42.384+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SecurityManager: Changing modify acls to: default
[2025-02-17T19:46:42.384+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SecurityManager: Changing view acls groups to:
[2025-02-17T19:46:42.384+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SecurityManager: Changing modify acls groups to:
[2025-02-17T19:46:42.385+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: default; groups with view permissions: EMPTY; users with modify permissions: default; groups with modify permissions: EMPTY
[2025-02-17T19:46:42.536+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO Utils: Successfully started service 'sparkDriver' on port 44023.
[2025-02-17T19:46:42.561+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SparkEnv: Registering MapOutputTracker
[2025-02-17T19:46:42.589+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SparkEnv: Registering BlockManagerMaster
[2025-02-17T19:46:42.605+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-02-17T19:46:42.605+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-02-17T19:46:42.609+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-02-17T19:46:42.624+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-977555e6-39f3-4334-9d0b-5da583da85df
[2025-02-17T19:46:42.637+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-02-17T19:46:42.648+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-02-17T19:46:42.735+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-02-17T19:46:42.782+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-02-17T19:46:42.862+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark:7077...
[2025-02-17T19:46:42.894+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO TransportClientFactory: Successfully created connection to spark/172.18.0.4:7077 after 17 ms (0 ms spent in bootstraps)
[2025-02-17T19:46:42.957+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250217194642-0003
[2025-02-17T19:46:42.959+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250217194642-0003/0 on worker-20250217194246-172.18.0.2-43155 (172.18.0.2:43155) with 1 core(s)
[2025-02-17T19:46:42.962+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO StandaloneSchedulerBackend: Granted executor ID app-20250217194642-0003/0 on hostPort 172.18.0.2:43155 with 1 core(s), 2.0 GiB RAM
[2025-02-17T19:46:42.962+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250217194642-0003/1 on worker-20250217194246-172.18.0.2-43155 (172.18.0.2:43155) with 1 core(s)
[2025-02-17T19:46:42.963+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO StandaloneSchedulerBackend: Granted executor ID app-20250217194642-0003/1 on hostPort 172.18.0.2:43155 with 1 core(s), 2.0 GiB RAM
[2025-02-17T19:46:42.965+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 33597.
[2025-02-17T19:46:42.966+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO NettyBlockTransferService: Server created on c20563c2c1e7:33597
[2025-02-17T19:46:42.967+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-02-17T19:46:42.971+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, c20563c2c1e7, 33597, None)
[2025-02-17T19:46:42.975+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO BlockManagerMasterEndpoint: Registering block manager c20563c2c1e7:33597 with 434.4 MiB RAM, BlockManagerId(driver, c20563c2c1e7, 33597, None)
[2025-02-17T19:46:42.978+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, c20563c2c1e7, 33597, None)
[2025-02-17T19:46:42.981+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, c20563c2c1e7, 33597, None)
[2025-02-17T19:46:42.992+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250217194642-0003/1 is now RUNNING
[2025-02-17T19:46:42.993+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:42 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250217194642-0003/0 is now RUNNING
[2025-02-17T19:46:43.174+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:43 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-02-17T19:46:43.382+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:43 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-02-17T19:46:43.391+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:43 INFO SharedState: Warehouse path is 'file:/usr/local/app/spark-warehouse'.
[2025-02-17T19:46:44.404+0000] {spark_submit.py:641} INFO - Traceback (most recent call last):
[2025-02-17T19:46:44.405+0000] {spark_submit.py:641} INFO - File "/usr/local/app/spark_applications/user_activity_consuming.py", line 36, in <module>
[2025-02-17T19:46:44.408+0000] {spark_submit.py:641} INFO - main()
[2025-02-17T19:46:44.409+0000] {spark_submit.py:641} INFO - File "/usr/local/app/spark_applications/user_activity_consuming.py", line 21, in main
[2025-02-17T19:46:44.411+0000] {spark_submit.py:641} INFO - .load()
[2025-02-17T19:46:44.412+0000] {spark_submit.py:641} INFO - ^^^^^^
[2025-02-17T19:46:44.413+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/streaming/readwriter.py", line 304, in load
[2025-02-17T19:46:44.413+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-02-17T19:46:44.414+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-02-17T19:46:44.422+0000] {spark_submit.py:641} INFO - pyspark.errors.exceptions.captured.AnalysisException: Failed to find data source: kafka. Please deploy the application as per the deployment section of Structured Streaming + Kafka Integration Guide.
[2025-02-17T19:46:44.467+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO SparkContext: Invoking stop() from shutdown hook
[2025-02-17T19:46:44.467+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-02-17T19:46:44.476+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO SparkUI: Stopped Spark web UI at http://c20563c2c1e7:4040
[2025-02-17T19:46:44.480+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-02-17T19:46:44.483+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-02-17T19:46:44.495+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-02-17T19:46:44.505+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO MemoryStore: MemoryStore cleared
[2025-02-17T19:46:44.506+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO BlockManager: BlockManager stopped
[2025-02-17T19:46:44.510+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-02-17T19:46:44.512+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-02-17T19:46:44.520+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO SparkContext: Successfully stopped SparkContext
[2025-02-17T19:46:44.521+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO ShutdownHookManager: Shutdown hook called
[2025-02-17T19:46:44.521+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO ShutdownHookManager: Deleting directory /tmp/spark-1a312255-bcc5-41d3-ab09-20f0e6bb1e9c/pyspark-f015cce7-ddbd-47cf-a757-78b4c6d16a25
[2025-02-17T19:46:44.524+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO ShutdownHookManager: Deleting directory /tmp/spark-5fe2ac9f-7b85-42e2-a533-44e3ecfe1661
[2025-02-17T19:46:44.526+0000] {spark_submit.py:641} INFO - 25/02/17 19:46:44 INFO ShutdownHookManager: Deleting directory /tmp/spark-1a312255-bcc5-41d3-ab09-20f0e6bb1e9c
[2025-02-17T19:46:44.563+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077:7077 --conf spark.executor.memory=2g --executor-cores 1 --executor-memory 2g --driver-memory 1g --name arrow-spark --deploy-mode client /usr/local/app/spark_applications/user_activity_consuming.py. Error code is: 1.
[2025-02-17T19:46:44.567+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=kafka_streaming_process_pipeline, task_id=example_consumer, run_id=manual__2025-02-17T19:46:22.248865+00:00, execution_date=20250217T194622, start_date=20250217T194639, end_date=20250217T194644
[2025-02-17T19:46:44.576+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-17T19:46:44.577+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 71 for task example_consumer (Cannot execute: spark-submit --master spark://spark:7077:7077 --conf spark.executor.memory=2g --executor-cores 1 --executor-memory 2g --driver-memory 1g --name arrow-spark --deploy-mode client /usr/local/app/spark_applications/user_activity_consuming.py. Error code is: 1.; 943)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077:7077 --conf spark.executor.memory=2g --executor-cores 1 --executor-memory 2g --driver-memory 1g --name arrow-spark --deploy-mode client /usr/local/app/spark_applications/user_activity_consuming.py. Error code is: 1.
[2025-02-17T19:46:44.590+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-02-17T19:46:44.602+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-17T19:46:44.604+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
